dotCMS/src/integration-test/java/com/dotcms/vanityurl/business/VanityUrlAPITest.java

```diff
@@ -60,271 +61,255 @@ public static void prepare() throws Exception {
         /* Default user */
         user = APILocator.systemUser();
 
-        /* APIs initialization */
-        hostAPI = APILocator.getHostAPI();
         contentletAPI = APILocator.getContentletAPI();
         vanityUrlAPI = APILocator.getVanityUrlAPI();
         filtersUtil = FiltersUtil.getInstance();
 
-        /* Load Cache */
-        vanityUrlCache = CacheLocator.getVanityURLCache();
+        cache=CacheLocator.getVanityURLCache();
 
         /* Default variables */
-        defaultHost = hostAPI.findDefaultHost(user, false);
-        defaultLanguageId = APILocator.getLanguageAPI().getDefaultLanguage().getId();
+        defaultHost =  new SiteDataGen().nextPersisted();
+        defaultLanguage = APILocator.getLanguageAPI().getDefaultLanguage();
     }
 
+    
+    
     /**
      * Creates a vanity url but no publish it, gets it and checks there is not in cache.
      *
      * @see VanityUrlAPI
      */
     @Test
-    public void getWorkingVanityUrl() {
+    public void testing_the_from_contentlet_method() throws Exception {
+
+
+        long i = System.currentTimeMillis();
+        String title = "VanityURL" + i;
+        String site = defaultHost.getIdentifier();
+        String uri = "/test1_" + i;
+        String forwardTo = "/about-us/" + CMSFilter.CMS_INDEX_PAGE;
+        int action = 200;
+        int order = 1;
+
+        Contentlet contentlet1 = filtersUtil.createVanityUrl(title, site, uri,
+                forwardTo, action, order, defaultLanguage.getId());
+
+        VanityUrl vanity1 = vanityUrlAPI.fromContentlet(contentletAPI.find(contentlet1.getInode(), user, false));
+
+        Assert.assertNotNull(vanity1);
+        Assert.assertEquals(title, vanity1.getTitle());
+        Assert.assertEquals(site, vanity1.getSite());
+        Assert.assertEquals(uri, vanity1.getURI());
+        Assert.assertEquals(forwardTo, vanity1.getForwardTo());
+        Assert.assertEquals(action, vanity1.getAction());
+        Assert.assertEquals(order, vanity1.getOrder());
+
+    }
+    
+    
+    
+    
+    /**
+     * Creates a vanity url but no publish it, gets it and checks there is not in cache.
+     *
+     * @see VanityUrlAPI
+     */
+    @Test
+    public void working_vanities_do_not_resolve() throws Exception {
```

Comment:
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)